pkg/events/emit.go

```diff
@@ -0,0 +1,86 @@
+package events
+
+import (
+	"context"
+	"os"
+
+	v1alpha12 "github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode/v1alpha1"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func NewEventEmitter(client kubernetes.Interface, logger *zap.SugaredLogger) *EventEmitter {
+	return &EventEmitter{
+		client: client,
+		logger: logger,
+	}
+}
+
+type EventEmitter struct {
+	client kubernetes.Interface
+	logger *zap.SugaredLogger
+}
+
+func (e *EventEmitter) EmitMessage(repo *v1alpha12.Repository, loggerLevel zapcore.Level, message string) {
+	event := makeEvent(repo, message)
+	if event != nil && e.client != nil {
+		if _, err := e.client.CoreV1().Events(event.Namespace).Create(context.TODO(), event, metav1.CreateOptions{}); err != nil {
+			e.logger.Infof("Cannot create event: %s", err.Error())
+		}
+	}
+
+	//nolint
+	switch loggerLevel {
+	case zapcore.DebugLevel:
+		e.logger.Debug(message)
+	case zapcore.ErrorLevel:
+		e.logger.Error(message)
+	case zapcore.InfoLevel:
+		e.logger.Info(message)
+	case zapcore.WarnLevel:
+		e.logger.Warn(message)
+	}
+}
+
+func makeEvent(repo *v1alpha12.Repository, message string) *v1.Event {
+	var namespace, name string
+	if repo != nil {
+		name = repo.GetName() + "-"
+		namespace = repo.GetNamespace()
+	} else {
+		// maybe output to pipelines-as-code namespace in case we don't have repoMatched
```

Comment:
I think it's fine to keep in the controller logs and not emit in pac namespace if  we didn't find any match

Reply:
if repo is not matched the checking out the events would be easier tha logs ? wdyt
so emitting in those case also